### PR TITLE
Add support for the Philips Hue adapter. r=azasypkin

### PR DESCRIPTION
**Pull request by gmarty**
This patch:
* Add partial support for the Philips Hue adapter (getting values is not done yet)
* Relax the use of eslint `comma-dangle` in preparation of https://github.com/fxbox/app/issues/47
* Fix minor issue in `foxbox.js` (better warning when no boxes are found and only poll the box when the user is logged in)
* Remove the code related to the deprecated service states

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -24,7 +24,8 @@
     "no-use-before-define": ["error", { "functions": false, "classes": true }],
     "no-var": ["error"],
     "quotes": ["error", "single"],
-    "semi": ["error", "always"]
+    "semi": ["error", "always"],
+    "comma-dangle": ["warn", "only-multiline"]
```
Comment (azasypkin) [inline .eslintrc]: __nit:__ Maybe let's enforce rule with ```"error"```?

Also I thought your ultimate goal was something more like ```["error", "always-multiline"]``` or I just misunderstood?

Comment (gmarty) [inline .eslintrc]: No that's correct. But for now I just want to allow it. We'll enforce this rule later when working on #47.

File: app/js/lib/foxbox/foxbox.js [hunk 2]
```diff
@@ -260,6 +267,10 @@ export default class Foxbox extends Service {
     }
 
     this[p.nextPollTimeout] = setTimeout(() => {
+      if (!this.isLoggedIn) {
```
Comment (azasypkin) [inline app/js/lib/foxbox/foxbox.js]: __nit:__ maybe just add this new condition to the ```if``` branch above (```if (!this[p.isPollingEnabled] || this[p.nextPollTimeout] || !this.isLoggedIn) ```)? In this case we'll have it inside two main places at the very beginning (```schedulePoll``` and ```refreshServicesByPolling```)

File: app/js/lib/foxbox/foxbox.js [hunk 3]
```diff
@@ -350,44 +361,6 @@ export default class Foxbox extends Service {
       });
   }
 
-  /**
-   * Fetch the state of a service from the box.
-   *
-   * @param {string} id The ID of the service.
-   * @return {Promise}
-   */
-  getServiceState(id) {
```
Comment (azasypkin) [inline app/js/lib/foxbox/foxbox.js]: \0/ Then maybe remove related dead code from ```refreshServicesByPolling```?

File: app/js/lib/foxbox/foxbox.js [hunk 4]
```diff
@@ -473,6 +446,8 @@ export default class Foxbox extends Service {
     switch (operationKind) {
       case 'TakeSnapshot':
         return 'Unit';
+      case 'power':
```
Comment (azasypkin) [inline app/js/lib/foxbox/foxbox.js]: __note:__ Just wanted to note that lower case ```power``` looks suspicious :)

File: app/js/lib/foxbox/settings.js [hunk 5]
```diff
@@ -62,6 +62,8 @@ export default class Settings extends Model {
     return new Promise(resolve => {
       // @todo Remove only the items set here.
       storage.clear();
```
Comment (azasypkin) [inline app/js/lib/foxbox/settings.js]: __note:__ also not related to your patch, but it's funny that fake storage defined above doesn't have ```clear``` method, so with fake storage this promise will never resolve :)

File: app/js/views/service.jsx [hunk 6]
```diff
@@ -33,13 +34,16 @@ export default class Service extends BaseView {
...
         return (<CameraServiceView service={this.state.service}
-                                   foxbox={this.foxbox} />);
+                                   foxbox={this.foxbox}/>);
+      case 'philips_hue@link.mozilla.org':
+        return (<LightServiceView service={this.state.service}
+                                  foxbox={this.foxbox}/>);
       default:
         return (<DefaultServiceView service={this.state.service}
-                                    foxbox={this.foxbox} />);
+                                    foxbox={this.foxbox}/>);
```
Comment (azasypkin) [inline app/js/views/service.jsx]: __question:__ why do you remove space before ```/>```? I was usually adding them to separate it from the value of the last attribute (haven't checked w3c spec on this though, but browsers are not restrictive at all). Let's agree on the same approach! What do you prefer? :)

File: app/js/views/services-list-item.jsx [hunk 7]
```diff
@@ -5,24 +5,57 @@ export default class ServicesListItem extends React.Component {
...
+    const id = this.getOperationByAlias('power').id;
+
+    // Optimistic update.
+    this.setState({ on });
+
+    this.foxbox.performSetOperation({
+        id,
+        kind: 'power'
+      }, value)
+      .catch((error) => {
```
Comment (azasypkin) [inline app/js/views/services-list-item.jsx]: __nit:__ alignment is somewhat messy, parameter object is on the same level as ```.catch``` branch

File: app/js/views/services-list-item.jsx [hunk 8]
```diff
@@ -5,24 +5,57 @@ export default class ServicesListItem extends React.Component {
...
+        // Revert back to the previous value.
+        this.setState({ on: !on });
+        console.error(error);
```
Comment (azasypkin) [inline app/js/views/services-list-item.jsx]: __nit:__ would be nice to have more detailed error description (errors returned from taxonomy are a bit vague), like:
```
console.error('Error occurred while blah-blah light switch blah-blah', error);
```

File: app/js/views/services-list-item.jsx [hunk 9]
```diff
@@ -5,24 +5,57 @@ export default class ServicesListItem extends React.Component {
...
   handleLightOnChange(evt) {
-    // No optimistic update because some network issues can't be caught.
     const on = evt.target.checked;
+    const value = on ? 'On' : 'Off';
+    const id = this.getOperationByAlias('power').id;
+
+    // Optimistic update.
+    this.setState({ on });
+
+    this.foxbox.performSetOperation({
```
Comment (azasypkin) [inline app/js/views/services-list-item.jsx]: __question:__ mmm I don't get why can't you just directly pass here operation returned from ```getOperationByAlias```?

File: app/js/views/services/light.jsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    name = name.trim();
+    this.foxbox.setTag({ name })
+      .then(() => {
+        this.populateTags(); // Needed to get the newly added tag ID.
+      });
+  }
+
+  render() {
+    return (
+      <div className="app-view">
```
Comment (azasypkin) [inline app/js/views/services/light.jsx]: __issue:__ Ah, sorry I'll need to account for new BaseView, so your render should just render body (probably with ```app-view__fill-body``` class)

File: app/js/views/services/light.jsx [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+  onServiceStateChanged(service) {
+    if (service.id !== this.service.id) {
+      return;
+    }
+
+    this.service = service;
+    this.setState({ data: service.tags });
+  }
+
+  populateTags() {
```
Comment (azasypkin) [inline app/js/views/services/light.jsx]: Mm, we'll we have tag management functionality in Light service page?

**Comment by azasypkin** [comment]
Okay looks good to me! I don't have Hue to check the lights, so I was just looking at the code part + that default and camera services are working properly (with yesterdays foxbox).

__r+__ with nits handled, questions answered, Light service view fixed (it should be migrated to BaseView approach) and green Travis!

Thanks

File: app/js/lib/foxbox/foxbox.js (hunk 3 above)
Comment (gmarty) [inline app/js/lib/foxbox/foxbox.js]: I'll leave the code related to polling for now, but I'm planning on refactoring it for https://github.com/fxbox/app/pull/73.

File: app/js/views/service.jsx (hunk 6 above)
Comment (gmarty) [inline app/js/views/service.jsx]: I think I must have configured my IDE in the past to remove the whitespace before `/>`.
It doesn't change anything once the code is compiled from JSX to plain JS though.

File: app/js/lib/foxbox/foxbox.js (hunk 4 above)
Comment (gmarty) [inline app/js/lib/foxbox/foxbox.js]: Ah it's been renamed `LightOn` in the latest version.

File: app/js/views/services-list-item.jsx (hunk 9 above)
Comment (gmarty) [inline app/js/views/services-list-item.jsx]: I haven't realised I could do that.

File: app/js/lib/foxbox/settings.js (hunk 5 above)
Comment (gmarty) [inline app/js/lib/foxbox/settings.js]: True. I'm going to fix it in this PR or I'll just forget about it!

File: app/js/views/services/light.jsx (hunk 11 above)
Comment (gmarty) [inline app/js/views/services/light.jsx]: It's the fake tags for now, purely frontend. I'll keep them for now but they'll probably go away soon.

**Pull request merged**